bootstrap.example.toml

```diff
@@ -740,11 +740,19 @@
 # result (broken, compiling, testing) into this JSON file.
 #rust.save-toolstates = <none> (path)
 
-# This is an array of the codegen backends that will be compiled for the rustc
-# that's being compiled. The default is to only build the LLVM codegen backend,
-# and currently the only standard options supported are `"llvm"`, `"cranelift"`
-# and `"gcc"`. The first backend in this list will be used as default by rustc
-# when no explicit backend is specified.
+# This array serves three distinct purposes:
+# - Backends in this list will be automatically compiled and included in the sysroot of each
+# rustc compiled by bootstrap.
+# - The first backend in this list will be configured as the **default codegen backend** by each
+# rustc compiled by bootstrap. In other words, if the first backend is e.g. cranelift, then when
+# we build a stage 1 rustc, it will by default compile Rust programs using the Cranelift backend.
```

Comment:
Discussion: this feels not great, ideally I think we want to make sure the user actually specifies a default, if multiple cg backends are used 🤔

Reply:
(Note that this is pre-existing behavior)

Currently the array is a bit multi-purpose, but as long as we clearly document it and it is intuitive, I don't think it's an issue. In 99.9% cases if you want to make something the default backend, you probably also want to make it be built by default, and this gets you both of these. We could introduce a `default-codegen-backend` field or something, but I don't think it's worth the churn at the moment.